File: moderate/lucky_tickets.rb
def choose(n, k)
  (1..k).inject(1) { |a, e| a * (n - e + 1) / e }
end

File.open(ARGV[0]).each_line do |line|
  m = line.to_i
  r = 0
  (0...m / 2).each do |i|
    r += (-1)**i * choose(m, i) * choose(11 * m / 2 - 1 - 10 * i, m - 1)
  end
  puts r
end

File: moderate/lucky_tickets.pl
use strict;
use bigint;

open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my $r = 0;
    for ( my $i = 0 ; $i < $line / 2 ; $i++ ) {
        $r +=
          (-1)**$i *
          ( 0 + $line )->bnok($i) *
          ( 11 * $line / 2 - 1 - 10 * $i )->bnok( $line - 1 );
    }
    print "$r\n";
}
close(INFILE);
